spacy/pipeline/edit_tree_lemmatizer.py

```diff
@@ -160,27 +160,20 @@ def predict(self, docs: Iterable[Doc]) -> List[Ints2d]:
     def _scores2guesses(self, docs, scores):
         guesses = []
         for doc, doc_scores in zip(docs, scores):
-            if self.top_k == 1:
-                doc_guesses = doc_scores.argmax(axis=1).reshape(-1, 1)
-            else:
-                doc_guesses = np.argsort(doc_scores)[..., : -self.top_k - 1 : -1]
-
-            if not isinstance(doc_guesses, np.ndarray):
-                doc_guesses = doc_guesses.get()
-
+            if not isinstance(doc_scores, np.ndarray):
+                doc_scores = doc_scores.get()
             doc_compat_guesses = []
-            for token, candidates in zip(doc, doc_guesses):
-                tree_id = -1
-                for candidate in candidates:
+            for i, token in enumerate(doc):
+                for _ in range(self.top_k):
+                    candidate = int(doc_scores[i].argmax())
```

Comment:
I think some variation of quickselect has better complexity. This approach is O(n*k), while with quickselect you can do it with O(n) + O(k^2) or O(n) + O(k log k). Not sure how it compares in practice with all the memory movement. It's mostly an empirical question.